lib/src/main/java/io/feeeei/circleseekbar/CircleSeekBar.java

```diff
@@ -42,246 +53,297 @@
     private static final String INSTANCE_WHEEL_SCROLL_ONLY_ONE_CIRCLE = "wheel_scroll_only_one_circle";
 
     private Paint mWheelPaint;
-
     private Paint mReachedPaint;
-
     private Paint mReachedEdgePaint;
-
     private Paint mPointerPaint;
-
     private int mMaxProcess;
     private int mCurProcess;
     private float mUnreachedRadius;
-    private int mReachedColor, mUnreachedColor;
-    private float mReachedWidth, mUnreachedWidth;
+    private int mReachedColor;
+    private int mUnreachedColor;
+    private float mReachedWidth;
+    private float mUnreachedWidth;
     private boolean isHasReachedCornerRound;
     private int mPointerColor;
     private float mPointerRadius;
-
     private double mCurAngle;
-    private float mWheelCurX, mWheelCurY;
-
-    private boolean isHasWheelShadow, isHasPointerShadow;
-    private float mWheelShadowRadius, mPointerShadowRadius;
-
+    private float mWheelCurX;
+    private float mWheelCurY;
+    private boolean isHasWheelShadow;
+    private boolean isHasPointerShadow;
+    private float mWheelShadowRadius;
+    private float mPointerShadowRadius;
     private boolean isHasCache;
     private Canvas mCacheCanvas;
-    private Bitmap mCacheBitmap;
-
+    private PixelMap mCacheBitmap;
     private boolean isCanTouch;
-
     private boolean isScrollOneCircle;
-
     private float mDefShadowOffset;
-
     private OnSeekBarChangeListener mChangListener;
 
+    /**
+     * CircleSeekBar Constructor.
+     *
+     *  @param context - context for CircleSeekBar constructor
+     *
+     */
     public CircleSeekBar(Context context) {
         this(context, null);
+        setEstimateSizeListener(this);
+        addDrawTask(this::onDraw);
```

Comment:
Place all the setListener methods within a private function and call only the private method from the constructors to avoid code duplication.

Reply:
Done